### DISCUSSION TITLE
PLEASE HELP! Can’t push code to GitHub repository anymore.

### DISCUSSION THREAD

**Discussion by kxnzx** [community/community#27305] [Pull Requests] [answered]
PLEASE HELP! Can’t push code to GitHub repository anymore.

Nothing happens anymore when I push code to a new remote repository from the MacOS Terminal. It used to work just fine, but not since recently and I have no idea why. I use the following Command Lines:

- cd documents
- cd “name of file”
- git init
- git add . 
- git commit -m “Initial commit”
- git remote add origin https://github.com/username/nameoffile
- git push -u origin master 

<img width="891" alt="Screenshot" src="https://user-images.githubusercontent.com/85115289/181252255-cc2931b1-8d68-4f7b-8a2c-83e989c9faa0.png">

Everything goes well (git folder is created etc.) except for the Push to the new remote repo. Nothing happens. When I close the terminal I get the following message: 

“Do you want to end the active processes in this window? If you close this window, the active processes will be ended: git (3), git-remote-https.”

I also used VSCode to commit and push, but this also does not work. It shows that it’s syncing, but it goes on forever and nothing is Pushed. 

The things I did to try to solve the problem:

- Setting Username and Email with: git config --global user.name “” and git config --global user.email “"
- Updating Git. I had Git version 2.30.1 (Apple Git -130) so I made an update with the Tim Harper Installer for Git. Now I have Git version 2.33.0

After the Git update I tried to push code again via the Terminal, but again nothing happens. After the push I now receive a Git-Credential-Osxkeychain window on which you can allow a password (which I disallowed). 

<img width="378" alt="Schermafbeelding 2022-07-27 om 14 59 06" src="https://user-images.githubusercontent.com/85115289/181254230-a27427be-2074-416d-8016-dfb4b01a2983.png">

On top of that I got a deprecation warning:

<img width="1264" alt="Schermafbeelding 2022-07-27 om 15 00 53" src="https://user-images.githubusercontent.com/85115289/181253338-c7521c2d-f036-4139-96a6-6bd34a782df4.png">

To surpress the deprecation warning I used the following Command Lines: 

-  echo ${SHELL}

Response from the command:  /bin/zsh 

- echo 'export TK_SILENCE_DEPRECATION=1' >> ~/.zshrc
- Restart Terminal

Again I started over (deleted local git folder and new Github repo). Then I made a new Github repo and tried to make a new push via the Terminal. This time no deprecation warning, and still nothing happens. No push to new remote repo. And on top of the Git Credential Osxkeychain Password window I also receive this window (which I ignored both):

<img width="378" alt="Schermafbeelding 2022-07-27 om 15 08 13" src="https://user-images.githubusercontent.com/85115289/181254737-cec92326-7063-400f-9d0b-99dfbb799132.png">

When I close the Terminal I receive the following message again:  

“Do you want to end the active processes in this window? If you close this window, the active processes will be ended: …” 

After that I tried again, same thing nothing happens with the same message and password windows as above, but now I also receive this command response after the push: 

<img width="488" alt="Schermafbeelding 2022-07-27 om 14 28 04" src="https://user-images.githubusercontent.com/85115289/181251776-0bace7db-89ca-4e01-87f1-d96d6b302f99.png">

What should I do? I hope I can get help from the community. What is the cause of all of this? And most important what is the solution? 

Your help is much appreciated!!

PS. I don't know if I used the correct category. Please place it in the right category for me if I didn't. Thanks in Advance!!

**Comment by jfullstackdev** [comment]
you can try GitHub Desktop also even without Git,

push the commits but not on the default branch, so you can create a pull request on GitHub and they will be verified

**Reply by kxnzx** [reply]
Hello [devstatic](https://github.com/jdevstatic), thank you for your suggestion! I have no experience working with GitHub Desktop. I have looked into it, but it looks confusing to me.... I could not find a good tutorial or manual on this topic. What do you mean with "push the commits but not on the default branch"?

**Comment by jfullstackdev** [comment]
is this fixed by now ?

**Reply by kxnzx** [reply]
Hello [jdevstatic](https://github.com/jdevstatic), Unfortunately no, I still hope someone can help me solve the issue? Does anyone know? Help is much appreciated!

**Comment by SamNickGammer** [comment]
You Need to add Auth Token to your Local GIT System

**Reply by kxnzx** [reply]
Hello SamNickGammer, thank you for your message. Can you please read my latest post below and elaborate? I have new screenshots. I tried to push again and I still receive the same message when closing the terminal “Do you want to end the active processes in this window? If you close this window, the active processes will be ended: git (2), git-remote-https, Wish, sh."

**Comment by jfullstackdev** [comment]
here is what I'm saying about GitHub Desktop

download it, then clone the target repo

![image](https://user-images.githubusercontent.com/47092464/181683153-c4243b38-d69b-471b-ae8c-a55ece04604e.png)

![image](https://user-images.githubusercontent.com/47092464/181683196-5ddbfe56-daab-4e07-80ac-30bb692cdb60.png)

you can choose any directory you want

then 

![image](https://user-images.githubusercontent.com/47092464/181683284-76395477-e597-4915-91b6-729a3c26cfda.png)

click Show in Explorer, then there is the File Explorer, drag all the change files from your local, say mine is 

![image](https://user-images.githubusercontent.com/47092464/181683457-4e3e6bc5-eab8-40fc-b6e1-4a79c910ca38.png)

close the folder, you will see changes once you click on any white space,

![image](https://user-images.githubusercontent.com/47092464/181683535-31412ef8-a761-4bff-8a21-1fa5ed52fdd1.png)

then you need to commit it, in my case, to main, but since I did not setup my commit signature that will be unverified, but the same as GitHub Web, you can create a new branch, that way it will be done through pull request, 

but going back, commit it 

![image](https://user-images.githubusercontent.com/47092464/181683798-d07c81c7-236d-4a04-8cdf-4aecf7420e95.png)

then click Publish, since this branch is not yet publish on GitHub

**Reply by kxnzx** [reply]
Thank you [jdevstatic](https://github.com/jdevstatic) for the instructions. I appreciated it! I am going to look into Github Desktop as a last resort, because I prefer working with the command line. 

My question regarding my issue remains unanswered unfortunately. I really want to solve the issue. Still hope someone can help me out... I restarted my macbook, started fresh and tried to push again via the Terminal. Again I received this authentification window:

<img width="993" alt="Schermafbeelding 2022-07-29 om 12 40 29" src="https://user-images.githubusercontent.com/85115289/181742669-d3e0618c-435a-43fb-90c8-61fe7435d378.png">

I clicked "No". Then nothing happens. When I closed the terminal I received the same message again: 

<img width="993" alt="Schermafbeelding 2022-07-29 om 12 45 10" src="https://user-images.githubusercontent.com/85115289/181743986-df0ef84f-a3df-4d3e-bf7b-72757693c364.png">

Translation: “Do you want to end the active processes in this window? If you close this window, the active processes will be ended: git (2), git-remote-https, Wish, sh."

"Wish" and "sh" are new, they showed up since I updated my Git. What does it mean? 
What to do to solve this issue?

**Comment by Mamulashvili** [comment]
Have you tried github personal access token like described here: 
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token  ?

**Reply by kxnzx** [reply]
Hello [Mamulashvili](https://github.com/Mamulashvili), Thank you for the link. I have read the documentation. I have never been prompted for my username and password before, because my credentials were not cached on my computer. For that reason I followed these instructions first: https://docs.github.com/en/get-started/getting-started-with-git/caching-your-github-credentials-in-git

- In have installed Brew and Git Credential Manager successfully. 
- After that I have set my username and password in keychain. 
- I have not set a personal access token (yet), because I first wanted to try to push again via the command line to see what will happen this time:

Everything went well and I accessed the GitHub authentication process successfully. I receive a command response that the push is Done:

<img width="449" alt="Schermafbeelding 2022-07-29 om 15 59 09" src="https://user-images.githubusercontent.com/85115289/181777103-becd1a83-25f6-4561-aad6-c4d495751a88.png">

However nothing happens again. When I refresh the repo on GitHub there is no updated repo. And when I want to close the terminal I again receive the same message: “Do you want to end the active processes in this window? If you close this window, the active processes will be ended: git (3), git-remote-https."

I left the terminal open (for a long time) to see if the remote repo on GitHub updates, but still nothing....

Please correct me if I'm wrong, but I am not sure if this issue has something to do with the personal access token? It seems as if the terminal keeps freezing. What are these active processes that keep running in the background? Is there some way to find out what is happening? 

Still looking for a solution...hope someone can help me.

**Comment by thuongtruong1009** [comment, marked as answer]
Hi @kxnzx 
You can try ```git config --global credential.useHttpPath true```

**Reply by kxnzx** [reply]
Oh Wow! [thuongtruong1009](https://github.com/thuongtruong1009) It worked! Thank you so much!! Could you please elaborate to make me understand what the issue was?

**Reply by thuongtruong1009** [reply]
Well, that's because when you want to access GitHub's repositories through VS Code or anywhere. You must have a certificate of authenticity from GitHub. By default, it will be turned off for security reasons and to prevent unauthorized access from hackers. So to be able to connect the GitHub account from local to remote. You must be logged into your account and enabled authenticated connections to the GitHub remote.
Tips: You should enable 2FA 2-factor authentication for your GitHub account for added security
I'm glad I could be of help to you.
Thanks very much!

**Reply by kxnzx** [reply]
Thank you for the explanation and the tip! I Just three more questions to fully understand....Is this new? Because previously I was able to push from local to remote without any authentification. And is it also (going to be) required to use the personal access token? Can you please explain this command: git config --global credential.useHttpPath true What does it exactly mean?

**Reply by thuongtruong1009** [reply]
Oh, I will answer your 3 questions as follows:
* First: this is not new. It is available when you install git
* Second: Previously you could push from local to remote without any authentication because the first time you logged into your local Git, you gave them an account via GitHub or Git Bash... So Git will remember that account, and you can push code from local to remote without authentication because they understand that account is owned by you. If you later change your Git account or use multiple accounts at the same time, this authentication will help Git identify the account owner. You can check the authenticated identity account in [Window](https://cmatskas.com/how-to-update-your-git-credentials-on-windows/) or [Mac](https://stackoverflow.com/questions/20195304/how-do-i-update-the-password-for-git) or [like this](https://medium.com/codex/git-credentials-on-macos-caching-updating-and-deleting-your-git-credentials-8d22b6126533)
* Third, command ```git config --global credential.useHttpPath true```
To use Windows Credential Manager for Git, you need to configure the credential helper as follows:
```git config --global credential.helper wincred```

* If you have multiple GitHub accounts that you use for different repositories, then you should configure the credentials to use the full repository path (instead of just the domain, which is the default) :
```git config --global credential.useHttpPath true```

Thanks for your question.
Have a nice day

**Reply by kxnzx** [reply]
thuongtruong1009 Thank you! Much appreciated! Good day to you :)

**Reply by kxnzx** [reply]
Hello thuongtruong1009 , I have set a 2FA 2-factor authentication. Now I am having the same issue as last time. I tried git config --global credential.useHttpPath true again, but nothing happens. The first try I received the option to authenticate myself, that went well (it also asked me if I had updated the 2FA 2-factor authentication). Nothing has been pushed. Then I tried again, this time no option to authenticate myself. And also nothing is being pushed. PLEASE Help!! :S

The terminal response looks good, but nothing is being pushed to the remote repo:

<img width="697" alt="Schermafbeelding 2022-08-06 om 19 07 53" src="https://user-images.githubusercontent.com/85115289/183258980-9888533a-af59-480b-8ce3-ac9dcdf14755.png">

And again I receive the same message when I want to close the window of the terminal “Do you want to end the active processes in this window? If you close this window, the active processes will be ended: git (3), git-remote-https."

<img width="273" alt="Schermafbeelding 2022-08-06 om 19 07 59" src="https://user-images.githubusercontent.com/85115289/183258982-a2a26e71-99fb-47cd-8754-6b272af213ac.png">

UPDATE: 
I was lucky this time! Since the commit was (still) staged in VS Code Source Control I could push it from there (I was also asked to authenticate). The possibility to finish the push in VS Code  does not always happen though. 

Why is GIT hanging constantly? Even when I pushed via VS Code the terminal kept giving me the same message “Do you want to end the active processes in this window? If you close this window, the active processes will be ended: git (3), git-remote-https." when closing the terminal (since I left it open).

**Reply by nathdipankarmns** [reply]
> Hi @kxnzx You can try `git config --global credential.useHttpPath true`

Faced a similar problem as the original post and this solved the issue for me as well! Thanks!

**Reply by KZ2mc** [reply]
Thank you SO MUCH! That's why everyone hates Git. The thing just stalls silently and you have to spend hours to find out what the hell is wrong. Why doesn't it just display an error or something?!

**Comment by mehmetbybz** [comment]
https://docs.bnbchain.org/docs/beaconchain/transfer#web-wallet

**Comment by SMAshhar** [comment]
![Screenshot from 2022-10-01 20-15-16](https://user-images.githubusercontent.com/63975782/193417098-e43e278b-79d7-4fb1-b21f-8ed10ce07674.png)


What to do now? I can upload stuff manualy, but can't push via CLI or GitKRaken

**Reply by thuongtruong1009** [reply]
Oh, It seems that you access the wrong sh key. 
You can try to replace the remote URL with ```https://github.com/SMAshhar/Test.git``` or [Setup new SH key](https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/GitHub-SSH-Key-Setup-Config-Ubuntu-Linux)

**Reply by SMAshhar** [reply]
git push --set-upstream origin main
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

Exactly the same error :(. 
I have to mention this too that I was facing an issue during "push" activity since last night. When i called "git push" it, asked me to authorize. 
1. Gave my username and password. didn't worked. Authorization error. 
            "Username for 'https://github.com/SMAshhar/Test.git': SMAshhar
            Password for 'https://SMAshhar@github.com/SMAshhar/Test.git': 
            remote: Support for password authentication was removed on August 13, 2021.
            remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
            fatal: Authentication failed for 'https://github.com/SMAshhar/Test.git/'
            "
3. Googled, told me to creat personal access token, I created that token. Username being username, password being the token (as told in the tutorial).  Gave the error 403.
4. Followed the guide "https://stackoverflow.com/questions/7438313/pushing-to-git-returning-error-code-403-fatal-http-request-failed". which lead me where I am standing now
.

**Comment by leproduction** [comment]
100% success to push files to github. https://www.youtube.com/watch?v=V8WbIERFOJU

**Comment by so-uvik** [comment]
If you're still facing this problem, this video can help you out as it did for me. (First try to push and fail, then do the git remote set-url thing otherwise it gave me an error)
https://www.youtube.com/watch?v=bvi37IbJXS4
